src/windows/cheats.c

```diff
@@ -756,29 +865,55 @@ static void window_cheats_rides_mouseup(rct_window *w, int widgetIndex)
 	}
 }
 
+static void window_cheats_text_input(rct_window *w, int widgetIndex, char *text) {
+	if (text == NULL)
+		return;
+
+	if (w->page == WINDOW_CHEATS_PAGE_MONEY && widgetIndex == WIDX_MONEY_SPINNER) {
+		window_cheats_convert_money(CONVERT_STRING_TO_NUMBER, text);
+		window_invalidate(w);
+	}
+}
+
 static void window_cheats_update(rct_window *w)
 {
 	w->frame_no++;
 	widget_invalidate(w, WIDX_TAB_1 + w->page);
 
-	if (widget_is_pressed(w, WIDX_INCREASE_PARK_RATING) || widget_is_pressed(w, WIDX_DECREASE_PARK_RATING))
-		park_rating_spinner_pressed_for++;
-	else
-		park_rating_spinner_pressed_for = 0;
-	if (park_rating_spinner_pressed_for >= CHEATS_PARK_RATING_SPINNER_PAUSE)
-		if (!(w->frame_no % 3)){
-			if (widget_is_pressed(w, WIDX_INCREASE_PARK_RATING)){
-				park_rating_spinner_value = min(999, 10 * (park_rating_spinner_value / 10 + 1));
-				widget_invalidate_by_class(WC_CHEATS, WIDX_PARK_RATING_SPINNER);
-				if (get_forced_park_rating() >= 0)
-					set_forced_park_rating(park_rating_spinner_value);
-			} else if (widget_is_pressed(w, WIDX_DECREASE_PARK_RATING)){
-				park_rating_spinner_value = max(0, 10 * (park_rating_spinner_value / 10 - 1));
-				widget_invalidate_by_class(WC_CHEATS, WIDX_PARK_RATING_SPINNER);
-				if (get_forced_park_rating() >= 0)
-					set_forced_park_rating(park_rating_spinner_value);
+	if (w->page == WINDOW_CHEATS_PAGE_MISC) {
+		if (widget_is_pressed(w, WIDX_INCREASE_PARK_RATING) || widget_is_pressed(w, WIDX_DECREASE_PARK_RATING))
```

Comment:
This is not how you do the repeat button. There is a flag you enable to do all of this. Can't remember the name but I recently made the change in the rate input window so go check that.